### DISCUSSION TITLE
Blocking the video tab on google - possible?

### DISCUSSION THREAD

**Discussion by turtle-walrus** [proginosko/LeechBlockNG#434] [Q&A] [answered]
I have youtube blocked, but it's easy to get around it through the "videos" tab when you simply do a video search. The videos are embedded and don't open to youtube to be blocked. Is there any fix to this while keeping normal google searches accessible?

**Comment by proginosko** [comment, marked as answer]
Try entering this into the list of sites to block:

`google.com/search?*tbm=vid`

**Reply by turtle-walrus** [reply]
This worked! Thank you.

**Reply by turtle-walrus** [reply]
> Try entering this into the list of sites to block:
> 
> `google.com/search?*tbm=vid`

This isn't working for me anymore - it used to but has recently stopped. Can't find any other info on blocking the videos tab or outright deleting it from google. Please help!

**Reply by proginosko** [reply]
Looks like Google changed the format of their search URLs. Try this:

`google.com/search?*udm=7&`

**Reply by one-eyed-man** [reply]
> Looks like Google changed the format of their search URLs. Try this:
> 
> `google.com/search?*udm=7&`

This works for me on the videos tab, but what about videos that appear on the All tab of google results? When they have the double arrow in the top right you can expand and watch within the results:
![image](https://github.com/user-attachments/assets/0c8cf051-d5e7-4b79-a88d-bd8c01ff1e8f)

I noticed that when you expand the video the following gets added to the end of the URL - 
`#fpstate=ive&vld=cid:`...

I've tried adding the following to my block list but no luck:
`google.com/search?*fpstate`
`google.com/search?*fpstate=ive&`
`google.com/search?*#fpstate=ive&vld=cid:`

**Reply by proginosko** [reply]
Try using `google.com/search?**#fpstate` and deselect the option "Ignore fragment (hash part) when checking URL" under Advanced Options.

**Reply by one-eyed-man** [reply]
I added that and deselected ignore fragment, interestingly when I click on the video the countdown timer now pops up for a split second in the top left but then quickly disappears, and the time doesn't go down

https://github.com/user-attachments/assets/beba33b7-17d8-4879-8976-c621a90559ed

**Reply by proginosko** [reply]
Rats, you just discovered a bug I didn't know about. 😂 The bug is that the "Ignore fragment" option is supposed to apply to _both_ blocking _and_ counting time, but currently it only applies to the blocking. I'll have to fix that in the next release. Sorry, I can't think of a workaround for now. If you change your settings to completely block those pages (not just a time limit) then it should work fine.

**Reply by one-eyed-man** [reply]
> Rats, you just discovered a bug I didn't know about. 😂 The bug is that the "Ignore fragment" option is supposed to apply to _both_ blocking _and_ counting time, but currently it only applies to the blocking. I'll have to fix that in the next release. Sorry, I can't think of a workaround for now. If you change your settings to completely block those pages (not just a time limit) then it should work fine.

No worries! Absolutely no pressure, but do you know roughly when the next update might be?

**Comment by slepto** [comment]
I came here to ask a related question.  

I block youtube in a leechblock filter, too.  My issue is embedded YouTube videos in ChatGPT threads.  I'll be researching a question and ChatGPT will embed a Youtube video in its answer.  If I hit 'play', I can watch the video!  


Is there a way to block all embedded youtube videos, regardless of the website or domain that's presenting them?

**Reply by smayer97** [reply]
The ONLY effective way I have ever found to block ALL embedded YouTube videos in ANY website is via the Windows HOST file. Anything else does not work effectively. You simply add/remove an entry for youtube.com (and possibly other variations). 

You can do it manually or use tools to manage this file. I combine this with the Windows Task Scheduler so I can disable at specific times by automating the change to the HOST file to block/unblock YouTube. 

Some info on this is here: https://superuser.com/questions/1337287/blocking-all-youtube-videos-including-those-embedded-in-websites

I use a free tool called BlueLife Hosts Editor by sordum. It has command line functions so it can be easily integrated with Task Scheduler to alter the HOST file on demand.
https://www.sordum.org/8266/bluelifehosts-editor-v1-6/

One caveat: DNS Over HTTPS ignores host files, so this will not work under that scenario.

**Reply by proginosko** [reply]
You can use the extension [uBlock Origin](https://ublockorigin.com/) to block embedded YouTube videos. Go to "My filters" in the settings and add these two filter rules:

`||youtube.com/embed/*$subdocument`
`||youtube-nocookie.com/embed/*$subdocument`

**Reply by smayer97** [reply]
Depending on how robust a solution you want/need, you might need to consider blocking access to enabling/disabling extensions. Start by turn this feature on in LeechBlock. Then look at solutions that prevent disabling extensions via other means.

Still, the most robust and simplest way to block all embedded YouTube videos is via the Windows Host file.

**Reply by proginosko** [reply]
> Still, the most robust and simplest way to block all embedded YouTube videos is via the Windows Host file.

Agreed, if you're using Windows.
